src/main/java/ru/yandex/practicum/filmorate/storage/dao/FeedDbStorage.java

```diff
@@ -0,0 +1,66 @@
+package ru.yandex.practicum.filmorate.storage.dao;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+import ru.yandex.practicum.filmorate.enums.EventType;
+import ru.yandex.practicum.filmorate.enums.Operation;
+import ru.yandex.practicum.filmorate.model.Event;
+import ru.yandex.practicum.filmorate.model.User;
+import ru.yandex.practicum.filmorate.storage.interfaces.IFeedStorage;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Repository
+public class FeedDbStorage implements IFeedStorage {
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public FeedDbStorage(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public void addEvent(Event event) {
+        SimpleJdbcInsert insert = new SimpleJdbcInsert(jdbcTemplate).withTableName("FEED")
+                .usingGeneratedKeyColumns("id");
+
+        insert.execute(Map.of("timestamp", event.getTimestamp(), "user_id", event.getUserId(),
+                "operation", event.getOperation(), "event_type", event.getEventType(),
+                "event_id", event.getEventId(), "entity_id", event.getEntityId()));
+        int id = jdbcTemplate.query("SELECT COUNT(ID) FROM FEED",
+                (rs, rowNum) -> rs.getInt("COUNT(ID)")).get(0);
+        event.setEventId(id);
```

Comment:
не заметил где это пригодилось. FeedDbStorage.addEvent вызывается в самом конце методов FeedService и можно проигнорить сгенерированный id и не выставлять его объекту Event

```suggestion
event.setEventId(insert.execute(...));
```

или можно так (вроде execute возвращает сгенерированный базой id)

Reply:
> или можно так (вроде execute возвращает сгенерированный базой id)

не, execute не вернет айдишник, там нужен executeAndReturnKey
но в целом я согласен, проще убрать установку айдишника, тк он не нужен вообще =)

Reply:
id event'a нужен для тестов.